platform/darwin/src/NSPredicate+MGLAdditions.mm

```diff
@@ -294,13 +294,56 @@ + (instancetype)mgl_predicateWithJSONObject:(id)object {
         return [NSPredicate predicateWithValue:YES];
     }
     if ([op isEqualToString:@"all"]) {
-        NSArray *subpredicates = MGLSubpredicatesWithJSONObjects([objects subarrayWithRange:NSMakeRange(1, objects.count - 1)]);
+        NSArray *jsonObjects = [objects subarrayWithRange:NSMakeRange(1, objects.count - 1)];
+        NSArray<NSPredicate *> *subpredicates = MGLSubpredicatesWithJSONObjects(jsonObjects);
+        if (jsonObjects.count == 2) {
+            // Determine if the expression is of BETWEEN type
+            if ([jsonObjects[0] isKindOfClass:[NSArray class]] &&
+                [jsonObjects[1] isKindOfClass:[NSArray class]]) {
+                NSArray *leftCondition = jsonObjects[0];
+                NSArray *rightCondition = jsonObjects[1];
+                NSString *leftOperator = leftCondition.firstObject;
+                NSString *rightOperator = rightCondition.firstObject;
+                
+                NSArray *limits;
+                NSExpression *leftConditionExpression;
+                
+                if([leftOperator isEqualToString:@">="] && [rightOperator isEqualToString:@"<="]) {
```

Comment:
Compare `leftPredicate.predicateOperatorType == NSGreaterThanOrEqualToPredicateOperatorType` instead of comparing strings. I don’t think there would be a significant performance difference either way, but we’d benefit from stronger type checking.